config/internal/configsource/component.go

```diff
@@ -21,8 +21,16 @@ import (
 
 // ErrSessionClosed is returned by WatchForUpdate functions when its parent Session
 // object is closed.
+// This error can be wrapped with additional information. Callers trying to identify this
+// specific error must use errors.Is.
 var ErrSessionClosed = errors.New("parent session was closed")
 
+// ErrValueUpdated is returned by WatchForUpdate functions when the value being watched
+// was changed or expired and needs to be retrieved again.
+// This error can be wrapped with additional information. Callers trying to identify this
+// specific error must use errors.Is.
+var ErrValueUpdated = errors.New("configuration must retrieve the updated value")
```

Comment:
Actually better to have an error for watcher not supported instead. Sorry for changing my mind

Reply:
np